app/lib/kafka.server.ts

```diff
@@ -44,36 +44,47 @@ export let send: (topic: string, value: string) => Promise<void>
 // testing.
 if (process.env.ARC_SANDBOX) {
   send = async (topic, value) => {
-    setOidcHttpOptions()
-    const producer = kafka.producer()
-    await producer.connect()
-    try {
-      await producer.send({ topic, messages: [{ value }] })
-    } finally {
-      await producer.disconnect()
+    if (kafka) {
+      setOidcHttpOptions()
+      const producer = kafka.producer()
+      await producer.connect()
+      try {
+        await producer.send({ topic, messages: [{ value }] })
+      } finally {
+        await producer.disconnect()
+      }
     }
   }
 } else {
   // FIXME: remove memoizee and use top-level await once we switch to ESM builds.
-  const getProducer = memoizee(
-    async () => {
-      setOidcHttpOptions()
-      const producer = kafka.producer()
-      await producer.connect()
-      ;['SIGINT', 'SIGTERM'].forEach((event) =>
-        process.once(event, async () => {
-          console.log('Disconnecting from Kafka')
-          await producer.disconnect()
-          console.log('Disconnected from Kafka')
-        })
-      )
-      return producer
-    },
-    { promise: true }
-  )
+  try {
+    const getProducer = memoizee(
+      async () => {
+        setOidcHttpOptions()
+        const producer = kafka.producer()
+        await producer.connect()
+        ;['SIGINT', 'SIGTERM'].forEach((event) =>
+          process.once(event, async () => {
+            console.log('Disconnecting from Kafka')
+            await producer.disconnect()
+            console.log('Disconnected from Kafka')
+          })
+        )
+        return producer
+      },
+      { promise: true }
+    )
 
-  send = async (topic, value) => {
-    const producer = await getProducer()
-    await producer.send({ topic, messages: [{ value }] })
+    send = async (topic, value) => {
+      const producer = await getProducer()
+      await producer.send({ topic, messages: [{ value }] })
+    }
```

Comment:
Why are are either of these in a `try`/`except` block? You are just declaring two functions.